### DISCUSSION TITLE
Building zeptoforth : howto ?

### DISCUSSION THREAD

**Discussion by manes6969** [tabemann/zeptoforth#48] [General]
Hi tabemann

I was trying to build zeptoforth , just for learning but I don't succeed very well. This is what I did 
-download the master code
-after unzip , in the root I excuted , make clean , make VERSION=1.0.3 , make install VERSION=1.03
-this creates indeed a bin folder with another subfolder RP2040
-there is a file zeptoforth_kernel-1.0.3.uf2 , but the size is only 169KB , whereas in your version 1.0.2 it is about 1.1MB.
-anyway I flashed it onto the RPI but get only an OK prompt and can type a few commands , but very limited.

I'm using Linux Mint 21

Can you tell me where I go wrong ?
Thanks in advance,
Kind regards,
Ronny

**Comment by tabemann** [comment]
There is more to building zeptoforth than that. If you want to build a full_usb build of zeptoforth, after flashing your Pico with the kernel UF2 file, execute the following:

```
utils/make_uf2_image.sh 1.0.3 rp2040 /dev/ttyUSB0 full_usb
```

Note that `1.0.3` should be whatever `VERSION` you specified earlier, and `/dev/ttyUSB0` should be whatever serial port's TTY device you are using. If you want a serial build rather than a USB build use `full` rather than `full_usb`. Note that this assumes that you have freshly flashed your Pico (or other RP2040-based board) with a kernel, or if not, that you have previously executed `erase-all` on your board. Also note that before you can actually use your board with USB, you need to power cycle it or execute `reboot` with screen or picocom or like against `/dev/ttyACM0` or whatever TTY device is being used for USB.

Note that this will also build the following UF2 file for the entire build at `bin/rp2040/1.0.3/zeptoforth_full_usb-1.0.3.uf2` (replace `1.0.3` with any other `VERSION` number). Note that building the UF2 file for the entire build may fail due to a checksum error if your machine is under too much load at the time.

Also, if you have no interest in creating such a UF2 file, you can execute the following instead:

```
utils/codeload3.py -B 115200 -p /dev/ttyUSB0 serial src/rp2040/forth/setup_full_usb.fs
```

All of the same applies here as well.

**Reply by manes6969** [reply]
Hi tabemann ,
It seems to not work : this is the output 
ronny@ronny-MINIPC-PN50-E1:~/Downloads/zepto2/zeptoforth-master/utils$ sudo sh make_uf2_image.sh 1.0.3 rp2040 /dev/ttyUSB0 full_usb
[sudo] password for ronny:    
rm: cannot remove 'screenlog.0': No such file or directory
make_uf2_image.sh: 34: ./utils/codeload3.py: not found
make_uf2_image.sh: 35: ./utils/codeload3.py: not found
make_uf2_image.sh: 36: screen: not found
make_uf2_image.sh: 37: screen: not found
make_uf2_image.sh: 38: screen: not found
grep: screenlog.0: No such file or directory
grep: screenlog.0: No such file or directory
grep: screenlog.0: No such file or directory
^C

any idea ?
Ronny

**Reply by tabemann** [reply]
You are executing it within the `utills` directory itself, so it won't find `utils/codeload3.py` as that is relative to the parent zeptoforth directory. Everything in the directions, and the scripts themselves, assume you are running it from your outer zeptoforth directory, in this case `~/Downloads/zepto2/zeptoforth-master/`. Also, the scripts require GNU Screen to be installed; there should be packages available for it for any Linux or xBSD distribution.

Hope that helps,
Travis

**Comment by manes6969** [comment]
Hi Travis ,

I guess I have to set a path or something , because it doesn't find make_uf2_image , which is obvious because it isn't there in this directory.

ronny@ronny-MINIPC-PN50-E1:~/Downloads/zepto2$ sudo sh make_uf2_image.sh 1.0.3 rp2040 /dev/ttyUSB0 full_usb
[sudo] password for ronny:    
sh: 0: cannot open make_uf2_image.sh: No such file

I installed gnu screens with  sudo apt install screen

**Reply by tabemann** [reply]
You don't need to run `utils/make_uf2_image.sh` as `root` (you just need to make sure your serial TTY device has the proper permissions, either by ensuring you are in the right group or, as I do as I don't remember how to do that, `chmod`ing the serial TTY device to `a+rw`), and you need to run it in the `utils` directory from the base `zeptoforth` directory as I specified, as:

```
utils/make_uf2_image.sh 1.0.3 rp2040 /dev/ttyUSB0 full_usb
```

Travis

**Comment by manes6969** [comment]
Hi Travis,

I'm so sorry , but it really doesn't work :


ronny@ronny-MINIPC-PN50-E1:~/Downloads/zeptoforth/utils$ sh make_uf2_image.sh 1.0.3 rp2040 /dev/ttyUSB0 full_usb
make_uf2_image.sh: 34: ./utils/codeload3.py: not found
make_uf2_image.sh: 35: ./utils/codeload3.py: not found
grep: screenlog.0: No such file or directory
grep: screenlog.0: No such file or directory
grep: screenlog.0: No such file or directory
grep: screenlog.0: No such file or directory
grep: screenlog.0: No such file or directory
grep: screenlog.0: No such file or directory
grep: screenlog.0: No such file or directory

I think its better to wait for your 1.0.3 version I guess ...

**Reply by tabemann** [reply]
Ronny,

The problem is that you are executing (prompt included)

```
ronny@ronny-MINIPC-PN50-E1:~/Downloads/zeptoforth/utils$ sh make_uf2_image.sh 1.0.3 rp2040 /dev/ttyUSB0 full_usb
```

rather than

```
ronny@ronny-MINIPC-PN50-E1:~/Downloads/zeptoforth$ sh utils/make_uf2_image.sh 1.0.3 rp2040 /dev/ttyUSB0 full_usb
```

The latter should work.

The issue is that `utils/make_uf2_image.sh` expects the `PWD` to be the zeptoforth base directory but here you are executing it with `utils` as the `PWD`.

Hope that helps,
Travis

**Comment by tmsgthb** [comment]
Hi all!

I dont want open a new discussion, because my problem is the same: 
how can i create zeptoforth image (uf2) for rpi pico 2 on macOS 15.2 with usb support?

The building process (make clean ; make ; make install) seems to be without error,
because after building in the bin directory there are some files:
```
zeptoforth_kernel-1.9.2-dev.bin
zeptoforth_kernel-1.9.2-dev.elf
zeptoforth_kernel-1.9.2-dev.ihex
zeptoforth_kernel-1.9.2-dev.uf2
```

But after flashing the rpi (zeptoforth_kernel-1.9.2-dev.uf2), i cannot access it via usb modem.

What am i doing wrong or which file do i have to use?

Thanks for your help!

Ps: here is build log:
```
make clean
/Applications/Xcode.app/Contents/Developer/usr/bin/make -C src/stm32f407 clean
rm -f ../../obj/zeptoforth.stm32f407.o ../../zeptoforth.stm32f407.elf ../../zeptoforth.stm32f407.bin ../../zeptoforth.stm32f407.ihex ../../zeptoforth.stm32f407.list ../../zeptoforth.stm32f407.map
/Applications/Xcode.app/Contents/Developer/usr/bin/make -C src/stm32f411 clean
rm -f ../../obj/zeptoforth.stm32f411.o ../../zeptoforth.stm32f411.elf ../../zeptoforth.stm32f411.bin ../../zeptoforth.stm32f411.ihex ../../zeptoforth.stm32f411.list ../../zeptoforth.stm32f411.map
/Applications/Xcode.app/Contents/Developer/usr/bin/make -C src/stm32l476 clean
rm -f ../../obj/zeptoforth.stm32l476.o ../../zeptoforth.stm32l476.elf ../../zeptoforth.stm32l476.bin ../../zeptoforth.stm32l476.ihex ../../zeptoforth.stm32l476.list ../../zeptoforth.stm32l476.map
/Applications/Xcode.app/Contents/Developer/usr/bin/make -C src/stm32f746 clean
rm -f ../../obj/zeptoforth.stm32f746.o ../../zeptoforth.stm32f746.elf ../../zeptoforth.stm32f746.bin ../../zeptoforth.stm32f746.ihex ../../zeptoforth.stm32f746.list ../../zeptoforth.stm32f746.map
/Applications/Xcode.app/Contents/Developer/usr/bin/make -C src/rp2040 clean
make -C boot clean
rm -f ../../../obj/boot.rp2040.o ../../../boot.rp2040.map ../../../boot.rp2040.list ../../../boot.rp2040.elf ../../../boot.rp2040.bin ../../../boot_with_checksum.rp2040.bin ../../../boot.rp2040.ihex
rm -f ../../obj/zeptoforth.rp2040.o ../../zeptoforth.rp2040.elf ../../zeptoforth.rp2040.bin ../../zeptoforth.rp2040.uf2 ../../zeptoforth.rp2040.ihex ../../zeptoforth.rp2040.list ../../zeptoforth.rp2040.map
/Applications/Xcode.app/Contents/Developer/usr/bin/make -C src/rp2040_big clean
make -C ../rp2040/boot clean
rm -f ../../../obj/boot.rp2040.o ../../../boot.rp2040.map ../../../boot.rp2040.list ../../../boot.rp2040.elf ../../../boot.rp2040.bin ../../../boot_with_checksum.rp2040.bin ../../../boot.rp2040.ihex
rm -f ../../obj/zeptoforth.rp2040_big.o ../../zeptoforth.rp2040_big.elf ../../zeptoforth.rp2040_big.bin ../../zeptoforth.rp2040_big.uf2 ../../zeptoforth.rp2040_big.ihex ../../zeptoforth.rp2040_big.list ../../zeptoforth.rp2040_big.map
/Applications/Xcode.app/Contents/Developer/usr/bin/make -C src/rp2350 clean
make -C boot clean
rm -f ../../../obj/boot.rp2350.o ../../../boot.rp2350.map ../../../boot.rp2350.list ../../../boot.rp2350.elf ../../../boot.rp2350.bin ../../../boot.rp2350.ihex
rm -f ../../obj/zeptoforth.rp2350.o ../../zeptoforth.rp2350.elf ../../zeptoforth.rp2350.bin ../../zeptoforth.rp2350.uf2 ../../zeptoforth.rp2350.ihex ../../zeptoforth.rp2350.list ../../zeptoforth.rp2350.map
/Applications/Xcode.app/Contents/Developer/usr/bin/make -C src/common clean
rm -f kernel_info.s

make
/Applications/Xcode.app/Contents/Developer/usr/bin/make -C src/stm32f407
/Applications/Xcode.app/Contents/Developer/usr/bin/make -C ../common kernel_info
rm -f kernel_info.s
echo '        define_word "kernel-platform", visible_flag' > kernel_info.s
echo '_kernel_platform:' >> kernel_info.s
echo '        push {lr}' >> kernel_info.s
echo '        string "'stm32f407'"' >> kernel_info.s
echo '        pop {pc}' >> kernel_info.s
echo '        end_inlined' >> kernel_info.s
echo >> kernel_info.s
echo '        define_word "kernel-version", visible_flag' >> kernel_info.s
echo '_kernel_version:' >> kernel_info.s
echo '        push {lr}' >> kernel_info.s
echo '        string "'1.9.2-dev'"' >> kernel_info.s
echo '        pop {pc}' >> kernel_info.s
echo '        end_inlined' >> kernel_info.s
echo >> kernel_info.s
echo '        define_word "kernel-date", visible_flag' >> kernel_info.s
echo '_kernel_date:' >> kernel_info.s
echo '        push {lr}' >> kernel_info.s
echo '        string "'2025 Jan  4 Szo 10:58:23 CET'"' >> kernel_info.s
echo '        pop {pc}' >> kernel_info.s
echo '        end_inlined' >> kernel_info.s
echo >> kernel_info.s
mkdir -p ../../obj
arm-none-eabi-as -g -o ../../obj/zeptoforth.stm32f407.o zeptoforth.s
arm-none-eabi-ld ../../obj/zeptoforth.stm32f407.o -T zeptoforth.ld --cref -Map ../../zeptoforth.stm32f407.map -o ../../zeptoforth.stm32f407.elf
arm-none-eabi-objdump -D ../../zeptoforth.stm32f407.elf > ../../zeptoforth.stm32f407.list
arm-none-eabi-objcopy ../../zeptoforth.stm32f407.elf ../../zeptoforth.stm32f407.bin -O binary
arm-none-eabi-objcopy ../../zeptoforth.stm32f407.elf ../../zeptoforth.stm32f407.ihex -O ihex
/Applications/Xcode.app/Contents/Developer/usr/bin/make -C src/stm32f411
/Applications/Xcode.app/Contents/Developer/usr/bin/make -C ../common kernel_info
rm -f kernel_info.s
...skipping...
echo '_kernel_date:' >> kernel_info.s
echo '        push {lr}' >> kernel_info.s
echo '        string "'2025 Jan  4 Szo 10:58:24 CET'"' >> kernel_info.s
echo '        pop {pc}' >> kernel_info.s
echo '        end_inlined' >> kernel_info.s
echo >> kernel_info.s
mkdir -p ../../obj
arm-none-eabi-as -g -o ../../obj/zeptoforth.rp2350.o zeptoforth.s
arm-none-eabi-ld ../../obj/zeptoforth.rp2350.o -T zeptoforth.ld --cref -Map ../../zeptoforth.rp2350.map -o ../../zeptoforth.rp2350.elf
arm-none-eabi-objdump -D ../../zeptoforth.rp2350.elf > ../../zeptoforth.rp2350.list
arm-none-eabi-objcopy ../../zeptoforth.rp2350.elf ../../zeptoforth.rp2350.bin -O binary
arm-none-eabi-objcopy ../../zeptoforth.rp2350.elf ../../zeptoforth.rp2350.ihex -O ihex
/Applications/Xcode.app/Contents/Developer/usr/bin/make -C boot
arm-none-eabi-as -g -o ../../../obj/boot.rp2350.o boot.s
arm-none-eabi-ld ../../../obj/boot.rp2350.o -T boot.ld --cref -Map ../../../boot.rp2350.map -o ../../../boot.rp2350.elf
arm-none-eabi-objdump -D ../../../boot.rp2350.elf > ../../../boot.rp2350.list
arm-none-eabi-objcopy ../../../boot.rp2350.elf ../../../boot.rp2350.bin -O binary
arm-none-eabi-objcopy ../../../boot.rp2350.elf ../../../boot.rp2350.ihex -O ihex
./make_uf2.sh ../../boot.rp2350.bin ../../zeptoforth.rp2350.bin ../../src/rp2350/mini_dict_filler ../../zeptoforth.rp2350.uf2

make install
/Applications/Xcode.app/Contents/Developer/usr/bin/make -C src/stm32f407 install
mkdir -p ../../bin/1.9.2-dev/stm32f407
cp ../../zeptoforth.stm32f407.bin ../../bin/1.9.2-dev/stm32f407/zeptoforth_kernel-1.9.2-dev.bin
cp ../../zeptoforth.stm32f407.ihex ../../bin/1.9.2-dev/stm32f407/zeptoforth_kernel-1.9.2-dev.ihex
cp ../../zeptoforth.stm32f407.elf ../../bin/1.9.2-dev/stm32f407/zeptoforth_kernel-1.9.2-dev.elf
/Applications/Xcode.app/Contents/Developer/usr/bin/make -C src/stm32f411 install
mkdir -p ../../bin/1.9.2-dev/stm32f411
cp ../../zeptoforth.stm32f411.bin ../../bin/1.9.2-dev/stm32f411/zeptoforth_kernel-1.9.2-dev.bin
cp ../../zeptoforth.stm32f411.ihex ../../bin/1.9.2-dev/stm32f411/zeptoforth_kernel-1.9.2-dev.ihex
cp ../../zeptoforth.stm32f411.elf ../../bin/1.9.2-dev/stm32f411/zeptoforth_kernel-1.9.2-dev.elf
/Applications/Xcode.app/Contents/Developer/usr/bin/make -C src/stm32l476 install
mkdir -p ../../bin/1.9.2-dev/stm32l476
cp ../../zeptoforth.stm32l476.bin ../../bin/1.9.2-dev/stm32l476/zeptoforth_kernel-1.9.2-dev.bin
cp ../../zeptoforth.stm32l476.ihex ../../bin/1.9.2-dev/stm32l476/zeptoforth_kernel-1.9.2-dev.ihex
cp ../../zeptoforth.stm32l476.elf ../../bin/1.9.2-dev/stm32l476/zeptoforth_kernel-1.9.2-dev.elf
/Applications/Xcode.app/Contents/Developer/usr/bin/make -C src/stm32f746 install
mkdir -p ../../bin/1.9.2-dev/stm32f746
cp ../../zeptoforth.stm32f746.bin ../../bin/1.9.2-dev/stm32f746/zeptoforth_kernel-1.9.2-dev.bin
cp ../../zeptoforth.stm32f746.ihex ../../bin/1.9.2-dev/stm32f746/zeptoforth_kernel-1.9.2-dev.ihex
cp ../../zeptoforth.stm32f746.elf ../../bin/1.9.2-dev/stm32f746/zeptoforth_kernel-1.9.2-dev.elf
/Applications/Xcode.app/Contents/Developer/usr/bin/make -C src/rp2040 install
mkdir -p ../../bin/1.9.2-dev/rp2040
cp ../../zeptoforth.rp2040.uf2 ../../bin/1.9.2-dev/rp2040/zeptoforth_kernel-1.9.2-dev.uf2
cp ../../zeptoforth.rp2040.bin ../../bin/1.9.2-dev/rp2040/zeptoforth_kernel-1.9.2-dev.bin
cp ../../zeptoforth.rp2040.ihex ../../bin/1.9.2-dev/rp2040/zeptoforth_kernel-1.9.2-dev.ihex
cp ../../zeptoforth.rp2040.elf ../../bin/1.9.2-dev/rp2040/zeptoforth_kernel-1.9.2-dev.elf
/Applications/Xcode.app/Contents/Developer/usr/bin/make -C src/rp2040_big install
mkdir -p ../../bin/1.9.2-dev/rp2040_big
cp ../../zeptoforth.rp2040_big.uf2 ../../bin/1.9.2-dev/rp2040_big/zeptoforth_kernel-1.9.2-dev.uf2
cp ../../zeptoforth.rp2040_big.bin ../../bin/1.9.2-dev/rp2040_big/zeptoforth_kernel-1.9.2-dev.bin
cp ../../zeptoforth.rp2040_big.ihex ../../bin/1.9.2-dev/rp2040_big/zeptoforth_kernel-1.9.2-dev.ihex
cp ../../zeptoforth.rp2040_big.elf ../../bin/1.9.2-dev/rp2040_big/zeptoforth_kernel-1.9.2-dev.elf
/Applications/Xcode.app/Contents/Developer/usr/bin/make -C src/rp2350 install
mkdir -p ../../bin/1.9.2-dev/rp2350
cp ../../zeptoforth.rp2350.uf2 ../../bin/1.9.2-dev/rp2350/zeptoforth_kernel-1.9.2-dev.uf2
cp ../../zeptoforth.rp2350.bin ../../bin/1.9.2-dev/rp2350/zeptoforth_kernel-1.9.2-dev.bin
cp ../../zeptoforth.rp2350.ihex ../../bin/1.9.2-dev/rp2350/zeptoforth_kernel-1.9.2-dev.ihex
cp ../../zeptoforth.rp2350.elf ../../bin/1.9.2-dev/rp2350/zeptoforth_kernel-1.9.2-dev.elf
```

**Comment by tabemann** [comment]
On Sat, Jan 4, 2025 at 7:36 AM tmsgthb ***@***.***> wrote:

> Hi all!
>
> I dont want open a new discussion, because my problem is the same:
> how can i create zeptoforth image (uf2) for rpi pico 2 on macOS 15.2 with
> usb support?
>
> The building process (make clean ; make ; make install) seems to be
> without error,
> because after building in the bin directory there are some files:
>
> zeptoforth_kernel-1.9.2-dev.bin
> zeptoforth_kernel-1.9.2-dev.elf
> zeptoforth_kernel-1.9.2-dev.ihex
> zeptoforth_kernel-1.9.2-dev.uf2
>
> But after flashing the rpi (zeptoforth_kernel-1.9.2-dev.uf2), i cannot
> access it via usb modem.
>
> What am i doing wrong or which file do i have to use?
>
> Thanks for your help!
>
> make only builds the zeptoforth kernel, nothing more. The kernel does not
have USB support, or for that matter, support for much of anything. Rather,
it is solely meant as a core to build a zeptoforth system on top of.

Your first steps were correct. However, after you do that, you need to hook
your RPi Pico 2 up to a USB-serial dongle using GPIO's 0 and 1 (pins 1 and
2) and build the rest of zeptoforth.

Afterward, issue the following with your current directory set to the
zeptoforth source tree's base directory:

$ utils/make_uf2_image.sh 1.9.2-dev rp2350 <the device corresponding to
your USB-serial dongle> full_usb

This will use your RPi Pico 2 to build a full_usb build of zeptoforth,
which will be in the bin/1.9.2-dev/rp2350 directory. It will also be
freshly installed on your RPi Pico 2, but will respond to serial, not USB;
to get USB connect to it once your done building with a terminal emulator
such as picocom and issue REBOOT or press control-C. Just be aware that it
will take a bit, and at the end of the process there will be a while where
it will look like it is doing nothing but is actually downloading an image
from the RPi Pico 2. At the end it will also output a couple messages that
look like error messages (e.g. it may complain about a missing file) but
which do not affect the process.

Warning - I have never tried this on macOS, so your mileage my vary. This
will require Python 3, which will automatically use pip to install pySerial
locally. Also note that you will need to make sure you can access your
USB-serial port, which may require changing group membership or like.

Note, however - this is all unnecessary unless you want to use the bleeding
edge code in git or want to make your own source modifications. If you just
want the latest release build, download zeptoforth-1.9.1.tar.gz from the
Releases on GitHub (do *not* select the 'Source code (tar.gz)' link because
that will not include any binaries). That will include the full_usb UF2
file under bin/1.9.1/rp2350.

Travis

**Reply by tmsgthb** [reply]
Hi Travis!

> Your first steps were correct. However, after you do that, you need to hook
> your RPi Pico 2 up to a USB-serial dongle using GPIO's 0 and 1 (pins 1 and
> 2) and build the rest of zeptoforth.

It was a missing link in my head.
Thanks for a quick reply.

> Warning - I have never tried this on macOS, so your mileage my vary. This
> will require Python 3, which will automatically use pip to install pySerial...

Once I get my usb-serial dongle I will see about the mileage.

> edge code in git or want to make your own source modifications. If you just
> want the latest release build, download zeptoforth-1.9.1.tar.gz from the

Yes, I used to do it before.

Tamás

**Comment by tmsgthb** [comment]
Hi Trevis!

For communication i use [Pi Debug probe](https://www.raspberrypi.com/documentation/microcontrollers/debug-probe.html)'s USB to UART bridge. 
Looks like with it i can create own uf2 image on platform rp2350.
In ./bin/1.9.2-dev/rp2350/ dir appear these files:
```
zeptoforth_full_usb-1.9.2-dev.bin
zeptoforth_full_usb-1.9.2-dev.ihex
zeptoforth_full_usb-1.9.2-dev.minidict.bin
zeptoforth_full_usb-1.9.2-dev.minidict.ihex
zeptoforth_full_usb-1.9.2-dev.uf2
```
Thanks a lot for helping. :)

On platform rp2040 make_uf2_image.sh fails. Last line of building:
```
TX: \ Clear an alarm interrupt
TX: : clear-alarm-int { index -- }
TX: index validate-alarm
TX: index bit INTR !
TX: index timer-irq NVIC_ICPR_CLRPEND!
TX: ;
TX:
TX: \ Is the alarm set
TX: : alarm-set? { index -- set? }
TX: index validate-alarm
TX: index bit ARMED bit@
TX: ;
TX:
TX: continue-module timer-internal
TX:
TX: \ Initialize the systick override using TIMER0
TX: : init-systick-override ( -- )
TX: [: ( -- ticks )
TX: us-counter 100. ud/ d>s
TX: ;] systick-override-hook!
error ;] systick-override-hook! unable to parse: systick-override-hook!

Error file src/rp2040/forth/setup_full_usb.fs line 48: could not upload file
>>>  #include src/rp2040/forth/timer.fs
No screen session found.
```

Tamás

**Comment by tabemann** [comment]
On Sat, Jan 18, 2025 at 10:00 AM tmsgthb ***@***.***> wrote:

> Hi Trevis!
>
> For communication i use Pi Debug probe
> <https://www.raspberrypi.com/documentation/microcontrollers/debug-probe.html>'s
> USB to UART bridge.
> Looks like with it i can create own uf2 image on platform rp2350.
> In ./bin/1.9.2-dev/rp2350/ dir appear these files:
>
> zeptoforth_full_usb-1.9.2-dev.bin
> zeptoforth_full_usb-1.9.2-dev.ihex
> zeptoforth_full_usb-1.9.2-dev.minidict.bin
> zeptoforth_full_usb-1.9.2-dev.minidict.ihex
> zeptoforth_full_usb-1.9.2-dev.uf2
>
> Thanks a lot for helping. :)
>
> On platform rp2040 make_uf2_image.sh fails. Last line of building:
>
> TX: \ Clear an alarm interrupt
> TX: : clear-alarm-int { index -- }
> TX: index validate-alarm
> TX: index bit INTR !
> TX: index timer-irq NVIC_ICPR_CLRPEND!
> TX: ;
> TX:
> TX: \ Is the alarm set
> TX: : alarm-set? { index -- set? }
> TX: index validate-alarm
> TX: index bit ARMED bit@
> TX: ;
> TX:
> TX: continue-module timer-internal
> TX:
> TX: \ Initialize the systick override using TIMER0
> TX: : init-systick-override ( -- )
> TX: [: ( -- ticks )
> TX: us-counter 100. ud/ d>s
> TX: ;] systick-override-hook!
> error ;] systick-override-hook! unable to parse: systick-override-hook!
>
> Error file src/rp2040/forth/setup_full_usb.fs line 48: could not upload file
> >>>  #include src/rp2040/forth/timer.fs
> No screen session found.
>
> Tamás
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/tabemann/zeptoforth/discussions/48#discussioncomment-11876135>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABDMC2FIDNVL3L3ZZBXYMAT2LJ3DVAVCNFSM6AAAAABUTDUPYSVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTCOBXGYYTGNI>
> .
> You are receiving this because you commented.
>

Thanks for catching this! I have been recently working primarily on the
RP2350 so I hadn't caught this. I have made a fix which I am testing now,
and I will be committing and pushing my changes shortly.

Travis

**Comment by tabemann** [comment]
On Sat, Jan 18, 2025 at 11:06 AM Travis Bemann ***@***.***> wrote:

> On Sat, Jan 18, 2025 at 10:00 AM tmsgthb ***@***.***> wrote:
>
>> Hi Trevis!
>>
>> For communication i use Pi Debug probe
>> <https://www.raspberrypi.com/documentation/microcontrollers/debug-probe.html>'s
>> USB to UART bridge.
>> Looks like with it i can create own uf2 image on platform rp2350.
>> In ./bin/1.9.2-dev/rp2350/ dir appear these files:
>>
>> zeptoforth_full_usb-1.9.2-dev.bin
>> zeptoforth_full_usb-1.9.2-dev.ihex
>> zeptoforth_full_usb-1.9.2-dev.minidict.bin
>> zeptoforth_full_usb-1.9.2-dev.minidict.ihex
>> zeptoforth_full_usb-1.9.2-dev.uf2
>>
>> Thanks a lot for helping. :)
>>
>> On platform rp2040 make_uf2_image.sh fails. Last line of building:
>>
>> TX: \ Clear an alarm interrupt
>> TX: : clear-alarm-int { index -- }
>> TX: index validate-alarm
>> TX: index bit INTR !
>> TX: index timer-irq NVIC_ICPR_CLRPEND!
>> TX: ;
>> TX:
>> TX: \ Is the alarm set
>> TX: : alarm-set? { index -- set? }
>> TX: index validate-alarm
>> TX: index bit ARMED bit@
>> TX: ;
>> TX:
>> TX: continue-module timer-internal
>> TX:
>> TX: \ Initialize the systick override using TIMER0
>> TX: : init-systick-override ( -- )
>> TX: [: ( -- ticks )
>> TX: us-counter 100. ud/ d>s
>> TX: ;] systick-override-hook!
>> error ;] systick-override-hook! unable to parse: systick-override-hook!
>>
>> Error file src/rp2040/forth/setup_full_usb.fs line 48: could not upload file
>> >>>  #include src/rp2040/forth/timer.fs
>> No screen session found.
>>
>> Tamás
>>
>> —
>> Reply to this email directly, view it on GitHub
>> <https://github.com/tabemann/zeptoforth/discussions/48#discussioncomment-11876135>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/ABDMC2FIDNVL3L3ZZBXYMAT2LJ3DVAVCNFSM6AAAAABUTDUPYSVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTCOBXGYYTGNI>
>> .
>> You are receiving this because you commented.
>>
>
> Thanks for catching this! I have been recently working primarily on the
> RP2350 so I hadn't caught this. I have made a fix which I am testing now,
> and I will be committing and pushing my changes shortly.
>

I have now tested my change, committed it, and pushed it to both devel and
master.

Travis

**Reply by tmsgthb** [reply]
Hi Trevis!

Thanks.

```
welcome
Welcome to zeptoforth
Built for rp2040, version 1.9.2-dev, on 2025 Jan 18 Szo 19:45:26 CET
zeptoforth comes with ABSOLUTELY NO WARRANTY: for details type `license'
 ok
```

Tamás
